jetty-core/jetty-core-server/src/main/java/org/eclipse/jetty/core/server/handler/ContextHandler.java

```diff
@@ -750,6 +865,54 @@ public void run(Runnable task)
                 throw new RuntimeException(e);
             }
         }
+
+        @Override
+        public void execute(Runnable runnable)
+        {
+            getServer().getContext().execute(() -> run(runnable));
+        }
+
+        @Override
+        public <T> T decorate(T o)
+        {
+            // TODO cache factory lookup?
+            DecoratedObjectFactory factory = ContextHandler.this.getBean(DecoratedObjectFactory.class);
+            if (factory != null)
+                return factory.decorate(o);
+            factory = getServer().getBean(DecoratedObjectFactory.class);
+            if (factory != null)
+                return factory.decorate(o);
+            return o;
+        }
+
+        @Override
+        public void destroy(Object o)
```

Comment:
I do not get what this `destroy` contract is about. What does it mean to destroy a context? And what does it mean to have a destroyed context still registered as a bean?

Reply:
This is the destroy from the Decorator, for CDI integration etc.